Refactor pyrgg/graph_gen.py and pyrgg/functions.py

ivanovmg opened the pull request:
#### Reference Issues/PRs

Re-submission of previous PR
https://github.com/sepandhaghighi/pyrgg/pull/54
this time into ``dev`` branch.

#### What does this implement/fix? Explain your changes.

Refactor ``pyrgg/graph_gen.py`` and ``pyrgg/functions.py``

#### Any other comments?

sepandhaghighi commented on pyrgg/graph_gen.py:
```diff
@@ -307,18 +334,64 @@ def tsv_maker(
+            buf.write(string)
+
+
+def json_to_yaml(filename):
```
I think you duplicate this function, exactly the same function is in `functions.py`

ivanovmg replied:
Thank you. I removed the duplication.

sepandhaghighi commented on pyrgg/graph_gen.py:
```diff
@@ -307,18 +334,64 @@ def tsv_maker(
+        print("[Error] Bad Input File")
+
+
+def json_to_pickle(filename):
```
I think you duplicate this function, exactly the same function is in `functions.py`

ivanovmg replied:
Thank you. Changed that.

sepandhaghighi reviewed (changes requested):
Please update `CHANGELOG.md` ;-)

sepandhaghighi commented:
@sadrasabouri Please review this PR ;-)

sadrasabouri commented:
> @sadrasabouri Please review this PR ;-)

Ok

sadrasabouri commented on CHANGELOG.md:
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - `pyrgg.py` renamed to `graph_gen.py`
 - Other functions moved to `functions.py`
 - Test system modified
+- Refactor `graph_gen.py` and `functions.py` using more Pythonic patterns.
```
It would be better if you make your change logs atomic as you do in first PR comment, I mean something like this:
```
- `graph_gen.py` refactored
- `functions.py` refactored
```
```

ivanovmg replied:
Thank you. I have just done this.

sadrasabouri reviewed (changes requested):
Thank you @ivanovmg .
Nice job.
I've left you a comment about `CHANGELOG.md`.

sepandhaghighi commented on pyrgg/functions.py:
```diff
@@ -271,54 +278,60 @@ def get_input(input_func=input):
+        return input_filter(result_dict)
+
+
+def _update_using_first_menu(result_dict, input_func):
```
Please add docstring for this function

sepandhaghighi commented on pyrgg/functions.py:
```diff
@@ -271,54 +278,60 @@ def get_input(input_func=input):
+    return result_dict
+
+
+def _update_using_second_menu(result_dict, input_func):
```
Please add docstring for this function

sepandhaghighi commented on pyrgg/graph_gen.py:
```diff
@@ -167,40 +164,77 @@ def json_maker(
+    return edge_number
+
+
+def _write_to_json(buf, edge_dic, weight_dic):
```
Please add docstring for this function

sepandhaghighi commented on pyrgg/graph_gen.py:
```diff
@@ -167,40 +164,77 @@ def json_maker(
+    buf.write("\n\t\t]\n\t}\n}")
+
+
+def _write_nodes_to_json(buf, edge_dic):
```
Please add docstring for this function

sepandhaghighi commented on pyrgg/graph_gen.py:
```diff
@@ -307,18 +334,30 @@ def tsv_maker(
     return edge_number
 
 
+def _write_separated_file(buf, edge_dic, weight_dic, separator, prefix=''):
```
Please update docstring of this function as other docstrings format.

ivanovmg commented:
I added docstrings as requested and added the missing if statement. But as soon as the missing if-statement did not raise issues in tests is a concern.
